uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RouteContextSupportTest.java

```diff
@@ -0,0 +1,156 @@
+import org.jboss.tools.fuse.reddeer.utils.ProjectFactory;
+import org.jboss.tools.fuse.reddeer.wizard.CamelXmlFileWizard;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.eclipse.reddeer.swt.api.Shell;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.impl.button.FinishButton;
+import org.eclipse.reddeer.workbench.impl.shell.WorkbenchShell;
+
+/**
+ * @author fpospisI
+ */
+@RunWith(RedDeerSuite.class)
+public class RouteContextSupportTest {
+	public static final String PROJECT_NAME = "routeContextSupportTest";
+	public static final String ROUTES_NAME = "camel-routes.xml";
+	
+	/**
+	 * <p>
+	 * routeContext support test
+	 * </p>
+	 * <b>Steps</b>
+	 * <ol>
+	 * <li>Create new project: STANDALONE - KARAF - CBR_SPRING.</li>
+	 * <li>In source tab change XML to external <routeContext> "camel-routes.xml" reference.</li>
+	 * <li>Create new XML - camel-routes.xml.</li>
+	 * <li>Edit 'Source' in camel-routes.xml.</li>
+	 * <li>Check camel-routes.xml 'Design' tab for containing items setted up in source. </li>
+	 * <li>Run as Local Camel Context (without tests) and check for Fuse related errors.</li>
+	 * </ol>
+	 * 
+	 * @author fpospisi
+	 */
+	@Test
+	public void testRouteContextSupport() {
+		/*
+		 * Create new project.
+		 * */
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(STANDALONE).runtimeType(KARAF).template(ProjectTemplate.CBR_SPRING).create();
+		
+		/*
+		 * Change content in 'Source'tab for main file.
+		 */
+		CamelEditor.switchTab("Source");
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		SourceEditor editor = new SourceEditor();
+		editor.selectText(0, 999999);		
+		editor.insertText("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+				+ "<beans xmlns=\"http://www.springframework.org/schema/beans\"\n"
+				+ "    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"        http://www.springframework.org/schema/beans https://www.springframework.org/schema/beans/spring-beans.xsd        http://camel.apache.org/schema/spring https://camel.apache.org/schema/spring/camel-spring.xsd\">\n"
+				+ "    <import resource=\"camel-routes.xml\"/>\n"
+				+ "    <camelContext id=\"cbr-example-context\" xmlns=\"http://camel.apache.org/schema/spring\">\n"
+				+ "        <routeContextRef ref=\"ExternalRoutes\"/>\n"
+				+ "    </camelContext>\n"
+				+ "</beans>");			
+		editor.save();
+		
+		/*
+		 * Create new XML file 'route-context.xml'.
+		 */
+		CamelXmlFileWizard wizz = new CamelXmlFileWizard();
+		wizz.openWizard();
+		wizz.setName(ROUTES_NAME); 
+		new FinishButton().click();
+		
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		
+		/*
+		 * Change content in 'Source' tab for 'route-context.xml'.
+		 */
+		CamelEditor.switchTab("Source");
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		SourceEditor editor_route = new SourceEditor();
+		editor_route.selectText(0, 999999);		
+		editor_route.insertText("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```

Comment:
I would prefer to have XML code in file inside `resources` folder and then use method `EditorManipulator.copyFileContentToCamelXMLEditor` to handle the same thing. WDYT?

It is beneficial because 
- code will be cleaner and more readable
- if you will want to test manually these test workflow you won't have to rewrite context.xml files from code but only reuse existing files from `resources` folder